### PR TITLE
DROTH-2305 Private road information disappears from some links

### PR DESCRIPTION
**Pull request by FilPinto**
-Updated the fill topology for the private road information no based more on the change type
-Created new tests

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/RoadLinkService.scala [hunk 1]
```diff
@@ -1143,40 +1142,42 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
...
+          case _ =>
         }
       }
     }
 
-    val changesToBeProcessed = changes.filter(change => change.oldId != change.newId).sortWith(_.vvhTimeStamp < _.vvhTimeStamp)
+    val changesToBeProcessed = changes.filterNot(change => change.oldId == change.newId)
+      .filterNot(change => change.oldId.isDefined && change.newId.isEmpty && change.changeType != ChangeType.Removed.value)
+      .filterNot(change => change.oldId.isEmpty && change.newId.isDefined && change.changeType != ChangeType.New.value)
+      .sortWith(_.vvhTimeStamp < _.vvhTimeStamp)
```
Comment (ricardofmsimoes) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/RoadLinkService.scala]: Sugestion:
`hanges.filterNot{ change => 
	val isOldEqNew = change.oldId == change.newId
	val newIdAndNotToRemoved = change.oldId.isDefined && change.newId.isEmpty && change.changeType != ChangeType.Removed.value
	val NewAndNotToNew = change.oldId.isEmpty && change.newId.isDefined && change.changeType != ChangeType.New.value
	
	isOldEqNew || newIdAndNotToRemoved || NewAndNotToNew

}.sortWith(_.vvhTimeStamp < _.vvhTimeStamp)`

Comment (FilPinto) [reply digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/RoadLinkService.scala]: I agree with you except in the vals

```
val changesToBeProcessed = changes.filterNot(change => change.oldId == change.newId || 
(change.oldId.isDefined && change.newId.isEmpty && change.changeType != ChangeType.Removed.value) || 
(change.oldId.isEmpty && change.newId.isDefined && change.changeType != ChangeType.New.value))
      .sortWith(_.vvhTimeStamp < _.vvhTimeStamp)
```

Comment (marquesfm) [reply digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/RoadLinkService.scala]: For me, I prefer the @ricardofmsimoes approach but don't like the names of the variables in particular "NewAndNotToNew" variable starting with big letters.

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/service/RoadLinkServiceSpec.scala [hunk 2]
```diff
@@ -1316,7 +1316,6 @@ class RoadLinkServiceSpec extends FunSuite with Matchers with BeforeAndAfter {
       )
 
       val testUser = "test_user"
-      sqlu"""Insert into ROAD_LINK_ATTRIBUTES (ID, NAME, LINK_ID, VALUE, CREATED_BY) values (11111111, 'PRIVATE_ROAD_ASSOCIATION', 11111111, $dummyRoadAssociationNameNumber, $testUser)""".execute
       sqlu"""Insert into ROAD_LINK_ATTRIBUTES (ID, NAME, LINK_ID, VALUE, CREATED_BY) values (22222222, 'PRIVATE_ROAD_ASSOCIATION', 22222222, $dummyRoadAssociationNameNumber, $testUser)""".execute
       sqlu"""Insert into ROAD_LINK_ATTRIBUTES (ID, NAME, LINK_ID, VALUE, CREATED_BY) values (33333331, 'PRIVATE_ROAD_ASSOCIATION', 33333333, $dummyRoadAssociationNameNumber, $testUser)""".execute
       sqlu"""Insert into ROAD_LINK_ATTRIBUTES (ID, NAME, LINK_ID, VALUE, CREATED_BY) values (33333332, 'ADDITIONAL_INFO', 33333333, '2', $testUser)""".execute
```
Comment (marquesfm) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/service/RoadLinkServiceSpec.scala]: The Id's of this inserts doesn't count for anything, so don't put the ID equal to the LinkId, make confusion to read.
Suggestion:
```scala
      sqlu"""Insert into ROAD_LINK_ATTRIBUTES (ID, NAME, LINK_ID, VALUE, CREATED_BY) values (2l, 'PRIVATE_ROAD_ASSOCIATION', 22222222, $dummyRoadAssociationNameNumber, $testUser)""".execute

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/RoadLinkService.scala (hunk 1 above)
Comment (FilPinto) [reply digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/RoadLinkService.scala]: val isOldEqNew = change.oldId == change.newId
      val isNotToExpire = change.oldId.isDefined && change.newId.isEmpty && change.changeType != ChangeType.Removed.value
      val isNotToCreate = change.oldId.isEmpty && change.newId.isDefined && change.changeType != ChangeType.New.value

Names applied to the vals

**Pull request merged**